### PR TITLE
Now sorting groups by name.

### PR DESCRIPTION
**Pull request by bgsavage**
Brought in a comparator library now using that to create our comparators.  Created an assessment comparator so for assessments to be sorted by grade then name.

### CODE REVIEW

File: webapp/src/main/webapp/src/app/groups/results/group-results.component.ts [hunk 1]
```diff
@@ -223,14 +219,15 @@ export class GroupResultsComponent implements OnInit {
...
 
     // cancel api calls in flight.
-    for(let loading of assessmentsLoading){
+    for (let loading of assessmentsLoading) {
       loading.subscription.unsubscribe();
     }
 
-    this.assessmentsLoading = this.assessmentsLoading.filter(loading => this.selectedAssessments.some(selected => loading.assessment.id == selected.id ));;
+    this.assessmentsLoading = this.assessmentsLoading.filter(loading => this.selectedAssessments.some(selected => loading.assessment.id == selected.id));
+    ;
```
Comment (jtreuting) [inline webapp/src/main/webapp/src/app/groups/results/group-results.component.ts]: This some be removed

Comment (jtreuting) [inline webapp/src/main/webapp/src/app/groups/results/group-results.component.ts]: Typo via phone and it won't let me edit.  That should say.  This should be removed.

Comment (bgsavage) [inline webapp/src/main/webapp/src/app/groups/results/group-results.component.ts]: oops yes.

File: webapp/src/main/webapp/src/app/groups/groups.resolve.ts [hunk 2]
```diff
@@ -2,11 +2,15 @@ import { Injectable } from "@angular/core";
...
 import { Observable } from "rxjs";
+import { byString } from "@kourge/ordering/comparator";
 
 @Injectable()
 export class GroupsResolve implements Resolve<any> {
   resolve(route: ActivatedRouteSnapshot, state: RouterStateSnapshot): Observable<any>|Promise<any>|any {
-    return this.service.getGroups();
+    return this.service.getGroups().toPromise()
+      .then( groups => {
+        return groups.sort((group1, group2) => byString(group1.name, group2.name));
```
Comment (esotrope) [inline webapp/src/main/webapp/src/app/groups/groups.resolve.ts]: The ordering lib also has this syntax which i think is a little more fluent

return groups.sort(ordering(byString).on(group => group.name).compare);

Comment (bgsavage) [inline webapp/src/main/webapp/src/app/groups/groups.resolve.ts]: I agree, this is cleaner.  Just fyi, one nit from the compiler is we need to specify type T when calling `on`.

So in this case:  `groups.sort(ordering(byString).on<any>(group => group.name).compare);`

**Pull request merged**